### ISSUE TITLE
Following Quick Start guide fails as `cargo tauri dev` doesn't find correct target directory

### ISSUE THREAD

**Issue by malaire** [tauri-apps/tauri-docs#817]
Following the [Quick Start guide](https://tauri.app/v1/guides/getting-started/setup/html-css-js) fails with following error:

```
error: proc macro panicked
 --> src/main.rs:8:10
  |
8 |     .run(tauri::generate_context!())
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: The `devPath` configuration is set to `"../ui"` but this path doesn't exist

error: could not compile `app` due to previous error
```

Commands I used (on Debian 11):

    cargo install tauri-cli
    cargo tauri init
    ; giving ../ui for two last questions as told by the guide
    cargo tauri dev

**Comment by malaire** [comment]
ok, so `cargo tauri init` doesn't create all the needed files (unlike `cargo init`)  and requires creating `../ui/index.html` manually.

After adding that, I get another error:

```
thread '<unnamed>' panicked at 'failed to rename app: failed to rename `/path/to/project/src-tauri/target/debug/app` to `/path/to/project/src-tauri/target/debug/my-app`

Caused by:
    No such file or directory (os error 2)', <REDACTED>
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Note: `src-tauri/target` doesn't exist because I have configured my target directory to be elsewhere and apparently tauri is unable to find correct target directory.

**Comment by FabianLars** [comment]
So tbh, i don't consider this a valid issue, because the guide tells you to create these files: https://tauri.app/v1/guides/getting-started/setup/html-css-js/#create-the-frontend. `cargo tauri init` can't take care of that since you're free to do whatever you want with your frontend. Check out `create-tauri-app` if you prefer full base setups.

That aside, the second comment is interesting, can you explain/show your folder/file structure? If it can build your app, the renaming _should_ work too, but it looks like we may have overlooked something.

**Comment by malaire** [comment]
> So tbh, i don't consider this a valid issue, because the guide tells you to create these files

yes, I got confused about what steps are needed when I don't want to use npm etc. - just cargo.

> That aside, the second comment is interesting, can you explain/show your folder/file structure? If it can build your app, the renaming _should_ work too, but it looks like we may have overlooked something.

In my global `.cargo/config` I have set absolute path for `build.target-dir` (which overrides default `"target"`) to share build-artifacts between projects. So target files will not be in `src-tauri/target` but in a completely different location outside the project directory.

If `tauri` doesn't support such configuration then this should be documented.

ps. Adding local `.cargo/config` with just `build.target-dir = "target"` fixes this issue.
